### PR TITLE
Aws & Azure infrastructure mock tests

### PR DESCRIPTION
**Pull request by jnpacker**
Increases code coverage to %77+

* Fix some remove and destroy conditions
* Add Fake Failure interfaces to make sure all the code paths are tested

### CODE REVIEW

**Comment by jnpacker** [comment]
Added Azure infra tests.  This also exposed that the condition errors for the provider secrets and how a failure is displayed in Azure should be addressed.

* Added Azure
* Added two new failure condition results for Azure

File: pkg/controllers/azure_infra.go [hunk 1]
```diff
@@ -102,7 +110,11 @@ func (r *HypershiftDeploymentReconciler) destroyAzureInfrastructure(hyd *hypdepl
 		credentials,
 	)(ctx); err != nil {
 		log.Error(err, "there was a problem destroying infrastructure on the provider, retrying in 30s")
-		return ctrl.Result{RequeueAfter: 30 * time.Second}, err
+		return ctrl.Result{RequeueAfter: 30 * time.Second}, r.updateStatusConditionsOnChange(
+			hyd, hypdeployment.PlatformConfigured,
+			metav1.ConditionFalse,
+			err.Error(),
```
Comment (zhujian7) [inline pkg/controllers/azure_infra.go:116]: this condition is just been set above(line 103), so here we just want to record the error information as message field, right? 
And should we set this condition for the AWS too when destroying AWS infra failed?

Comment (jnpacker) [reply pkg/controllers/azure_infra.go:116]: Correct, we want to display the message.  Yes it should be in AWS as well.

File: pkg/controllers/infra_interface.go [hunk 2]
```diff
@@ -127,6 +128,8 @@ var _ InfraHandler = &FakeInfraHandler{}
 
 type FakeInfraHandler struct{}
 
+type FakeInfraHandlerFailure struct{}
```
Comment (jnpacker) [inline pkg/controllers/infra_interface.go:131]: @zhujian7 a question:
For AWS we have two calls out, infra and IAM. In Azure its only Infra.
To test when Infra returns error, and then when IAM returms error, I will need two separate types. Because
AwsInfraCreator needs to have an err when I want to test it's failure, but I need its error NIL when I test the AwsIamCreator failure.

Any ideas, so I don't have to introduce another type with identical func's except AwsInfraCreator err == nil and AwsIamCreator == errors.New

I wanted to avoid having to create 6x implementations for the new type, just to test an IAMCreator failure

File: pkg/controllers/azure_infra.go (hunk 1 above)
Comment (jnpacker) [reply pkg/controllers/azure_infra.go:116]: Done!

File: pkg/controllers/infra_interface.go (hunk 2 above)
Comment (zhujian7) [reply pkg/controllers/infra_interface.go:131]: Sorry, I don't have another solution, and I agree with you that it is not worth creating 6x implementations for the new type to test an IAMCreator failure.
